File: mobile-apps/alloy-ios/Alloy/HomeAssistant/HomeAssistantAuth.swift
import Foundation

/// Home Assistant long-lived access token handling. The token itself
/// lives in the Keychain (see `SecureStore`); this type just exposes
/// typed read/write + a helper to build auth headers.
struct HomeAssistantAuth: Sendable {
    static let tokenKey = "ha.llat"

    static func saveToken(_ token: String) throws {
        try SecureStore.shared.set(token, for: tokenKey)
    }

    static func loadToken() -> String? {
        SecureStore.shared.get(tokenKey)
    }

    static func clear() {
        try? SecureStore.shared.delete(tokenKey)
    }

    static func bearerHeader(_ token: String) -> (String, String) {
        ("Authorization", "Bearer \(token)")
    }
}

File: mobile-apps/alloy-android/app/src/main/kotlin/com/alloyremote/alloy/homeassistant/HomeAssistantAuth.kt
package com.alloyremote.alloy.homeassistant

import com.alloyremote.alloy.settings.SecureStore

/**
 * Home Assistant long-lived access token handling. The token itself
 * lives in EncryptedSharedPreferences via [SecureStore]; this object
 * just exposes typed read/write helpers.
 */
object HomeAssistantAuth {
    private const val TOKEN_KEY = "ha.llat"

    fun saveToken(token: String) {
        SecureStore.set(TOKEN_KEY, token)
    }

    fun loadToken(): String? = SecureStore.get(TOKEN_KEY)

    fun clear() {
        SecureStore.delete(TOKEN_KEY)
    }

    fun bearerHeader(token: String): Pair<String, String> =
        "Authorization" to "Bearer $token"
}
